doma-processor/src/main/java/org/seasar/doma/internal/apt/annot/Annotations.java

```diff
@@ -191,7 +193,21 @@ private ScopeClass newScopeClass(TypeMirror scopeType) {
     TypeElement typeElement = ctx.getMoreTypes().toTypeElement(scopeType);
     List<? extends Element> members = ctx.getMoreElements().getAllMembers(typeElement);
```

Comment:
The `getAllMembers` method seems to return the result in non-deterministic ordering.
So in my environment, `ScopeTest` fails.

Reply:
It was not reproduced in my Windows environment.
However, I was able to confirm that the order changes in the Mac environment.
Sorted by method name to pass the test.  https://github.com/domaframework/doma/pull/648/commits/695dad910859d13ebbbd875d438852bbb4c5bc04

The test code smells ominous, do you have any good ideas?

Reply:
I don't know how ominous, but I have an idea.
Split validation and generation.

- Create a new `org.seasar.doma.internal.apt.processor.ScopeMethodProcessor` class.
- Move `@ScopeMethod` related validation code from the `EntityMetaFactory` class to the `ScopeMethodProcessor` class. Throw an `AptIllegalStateException` when you find an invalid case about `@ScopeMethod` in the EntityMetaFactory class.
- In a new `ScopeMethodProcessorTest` class, test not generated code but validation logic.
- Remove failure test cases from the `ScopeTest` class. In other words, test only generated code in the `ScopeTest` class.
 
I think that the above changes make our code simple.

Reply:
What I find ominous is that the product code contains sort logic just to pass the test.
I want to separate verification and generation in another PR

Reply:
> What I find ominous is that the product code contains sort logic just to pass the test.

There is another way to use JUnit 5 features:
https://junit.org/junit5/docs/5.7.1/user-guide/#writing-tests-conditional-execution-os

However, I think it is acceptable to call `Collections.sort`.

> I want to separate verification and generation in another PR

OK.

Reply:
@orekyuu 

We split validation from generation in #660.
And we added some changes.
If you have any comments, please let us know.